dev/AccessControl/Microsoft.Windows.Security.AccessControl.cpp

```diff
@@ -0,0 +1,50 @@
+﻿#include <pch.h>
+#include "Microsoft.Windows.Security.AccessControl.h"
+#include "Microsoft.Windows.Security.AccessControl.SecurityDescriptorHelpers.g.cpp"
+#include "Security.AccessControl.h"
+#include <sddl.h>
+
+namespace winrt::Microsoft::Windows::Security::AccessControl::implementation
+{
+    wil::unique_hlocal_security_descriptor GetSecurityDescriptor(array_view<winrt::Microsoft::Windows::Security::AccessControl::AppContainerNameAndAccess const> accessRequests, hstring const& principalStringSid, uint32_t principalAccessMask, DWORD* sdLength)
+    {
+        wil::unique_any_psid sid;
+        if (!principalStringSid.empty())
+        {
+            winrt::check_hresult(::ConvertStringSidToSidW(principalStringSid.data(), &sid));
+        }
+
+        std::vector<::AppContainerNameAndAccess> rawAccessRequests;
+        rawAccessRequests.reserve(accessRequests.size());
+        for (auto&& request : accessRequests)
+        {
+            rawAccessRequests.push_back({ request.appContainerName.data(), request.accessMask });
```

Comment:
Nit: `emplace_back` ?

Reply:
It's a plain old struct without a constructor, using aggregate initialization. So, it doesn't work with `emplace_back`. https://stackoverflow.com/questions/8782895/why-doesnt-emplace-back-use-uniform-initialization